demo/basic/progress.html

```diff
@@ -0,0 +1,186 @@
+<!DOCTYPE html>
+<script src="../../dist/scroll-timeline.js"></script>
+
+<style>
+@scroll-timeline progressTimeline {
+  
+}
+
+body {
+  height: 3000px;
+  padding: 0px;
+  margin: 0px;
+}
+
+#progress {
+  position: sticky;
+  top: 0px;
+  width: 100%;
+  height: 10px;
+  background-color: red;
+  animation-timeline: progressTimeline;
+  animation: progress 2s linear;
```

Comment:
duration is not necessary.  It defaults to 'auto', which should then use the intrinsic iteration duration, which is calculated in order to fill the timeline duration.